lib/statsd/instrument.rb

```diff
@@ -360,18 +293,55 @@ def method_visibility(method)
     end
   end
 
-  attr_accessor :logger
-  attr_writer :singleton_client
+  class << self
+    extend Forwardable
 
-  extend Forwardable
+    # The logger to use in case of any errors. The logger is also used as default
+    # logger for the LoggerBackend (although this can be overwritten).
+    # @return [Logger]
+    # @see StatsD::Instrument::Backends::LoggerBackend
+    attr_accessor :logger
 
-  def singleton_client
-    @singleton_client ||= StatsD::Instrument::Environment.current.client
-  end
+    # The StatsD client that handles method calls on the StatsD singleton.
+    #
+    # E.g. a call to `StatsD.increment` will be handled by this client.
+    #
+    # @return [StatsD::Instrument::Client]
+    attr_writer :singleton_client
+
+    # The StatsD client that handles method calls on the StatsD singleton
+    # @return [StatsD::Instrument::Client]
+    def singleton_client
+      @singleton_client ||= StatsD::Instrument::Environment.current.client
+    end
+
+    # @!method measure(name, value = nil, sample_rate: nil, tags: nil, &block)
+    #   (see StatsD::Instrument::Client#measure)
+    #
+    # @!method increment(name, value = 1, sample_rate: nil, tags: nil)
+    #   (see StatsD::Instrument::Client#increment)
+    #
+    # @!method gauge(name, value, sample_rate: nil, tags: nil)
+    #   (see StatsD::Instrument::Client#gauge)
+    #
+    # @!method set(name, value, sample_rate: nil, tags: nil)
+    #   (see StatsD::Instrument::Client#set)
+    #
+    # @!method histogram(name, value, sample_rate: nil, tags: nil)
+    #   (see StatsD::Instrument::Client#histogram)
+    #
+    # @!method distribution(name, value = nil, sample_rate: nil, tags: nil, &block)
+    #   (see StatsD::Instrument::Client#distribution)
+    #
+    # @!method event(title, text, tags: nil, hostname: nil, timestamp: nil, aggregation_key: nil, priority: nil, source_type_name: nil, alert_type: nil) # rubocop:disable Metrics/LineLength
+    #   (see StatsD::Instrument::Client#event)
+    #
+    # @!method service_check(name, status, tags: nil, hostname: nil, timestamp: nil, message: nil)
+    #   (see StatsD::Instrument::Client#service_check)
```

Comment:
These need to be inside the `class << self` for yardoc to recognize them as class/singleton methods rather than instance methods.